packages/keymanager-server/src/impl.ts

```diff
@@ -80,7 +76,7 @@ export class KeymanagerApi implements Api {
       message?: string;
     }[];
   }> {
-    const interchange = (slashingProtectionStr as unknown) as Interchange;
+    const interchange = JSON.parse(slashingProtectionStr) as Interchange;
```

Comment:
@dadepo do you know why JSON.parse() was not necessary if `slashingProtectionStr` is supposed to be a string?

Reply:
So can you add a nice long comment explaining this above my change and do conditional parsing?

```ts
    const interchange: Interchange = typeof slashingProtectionStr === "string" ? JSON.parse(slashingProtectionStr) : slashingProtectionStr;
```

Reply:
I added a comment but I feel that the conditional parsing is not necessary since the endpoint is going to be called with the payload within the body of the request. 

Also the snippet you shared won't work. What will work will require changing the type for `slashingProtectionStr: SlashingProtectionData | Record<string, unknown>` and then have the conditional parsing as follows:

```
    const interchange = typeof slashingProtectionStr === "string"
        ? (JSON.parse(slashingProtectionStr) as Interchange)
        : ((slashingProtectionStr as unknown) as Interchange);
```

Which I think is a bit too much since payload will always come within the body